test/unit/spec/signaling/v2/cancelableroomsignalingpromise.js

```diff
@@ -197,131 +222,25 @@ describe('createCancelableRoomSignalingPromise', () => {
-          it('the CancelablePromise resolves to the newly-constructed RoomV2', () => {
-            const test = makeTest();
-            test.createAndOfferDeferred.resolve();
-            test.createAndOfferDeferred.promise.then(() => {
-              const identity = makeIdentity();
-              const sid = makeParticipantSid();
-              test.transport.emit('connected', {
-                participant: {
-                  sid: sid,
-                  identity: identity
-                },
-                options: {
-                  // eslint-disable-next-line camelcase
-                  signaling_region: 'foo'
-                }
-              });
-            });
-            return test.cancelableRoomSignalingPromise.then(room => {
-              assert.equal(test.room, room);
-            });
-          });
-        });
-
-        context('when the CancelablePromise has been canceled', () => {
-          it('the CancelablePromise rejects with a cancelation error', () => {
-            const test = makeTest();
-            test.createAndOfferDeferred.resolve();
-            test.createAndOfferDeferred.promise.then(() => {
-              const identity = makeIdentity();
-              const sid = makeParticipantSid();
-              test.transport.emit('connected', {
-                participant: {
-                  sid: sid,
-                  identity: identity
-                },
-                options: {
-                  // eslint-disable-next-line camelcase
-                  signaling_region: 'foo'
-                }
-              });
-              test.cancelableRoomSignalingPromise.cancel();
-            });
-            return test.cancelableRoomSignalingPromise.then(() => {
-              throw new Error('Unexpected resolution');
-            }, error => {
-              assert.equal(
-                'Canceled',
-                error.message);
-            });
-          });
-
-          it('calls .disconnect on the newly-constructed RoomV2', () => {
-            const test = makeTest();
-            test.createAndOfferDeferred.resolve();
-            test.createAndOfferDeferred.promise.then(() => {
-              const identity = makeIdentity();
-              const sid = makeParticipantSid();
-              test.transport.emit('connected', {
-                participant: {
-                  sid: sid,
-                  identity: identity
-                },
-                options: {
-                  // eslint-disable-next-line camelcase
-                  signaling_region: 'foo'
-                }
-              });
-              test.cancelableRoomSignalingPromise.cancel();
-            });
-            return test.cancelableRoomSignalingPromise.then(() => {
-              throw new Error('Unexpected resolution');
-            }, () => {
-              assert(test.room.disconnect.calledOnce);
-            });
-          });
-
-          it('does not call .disconnect on the Transport', () => {
```

Comment:
wondering why was this case there ? and why was it deleted? is that change in behavior?

Reply:
@makarandp0 These tests were making sure that if the CancelablePromise is canceled just after RoomV2 is created, then we should close the Transport and PeerConnectionManager. That is why we had a Promise within the CancelablePromise before. However, I noticed that in `lib/cancelableroompromise.js`, we do the same thing after `cancelableRoomSignalingPromise` is resolved. So, I just removed the redundant code so that we no longer need the Promise within the CancelablePromise.